### PR TITLE
fix(LocaleModal): esc key closes the Locale Modal filtering

### PR DESCRIPTION
**Pull request by guilhermelMoraes**
### Related Ticket(s)

Locale selector:- Keyboard - Select region modal dialog [#1769](https://github.com/carbon-design-system/ibm-dotcom-library/issues/1769)

### Description

If the user is filtering the region at the `LocaleModal` and press the Escape key on the keyboard it closes the filter.

### Changelog

**New**

- keydown eventListener in `LocaleModal` to detect the Escape key press.

### CODE REVIEW

File: packages/react/src/components/LocaleModal/LocaleModal.js [hunk 1]
```diff
@@ -71,6 +71,21 @@ const LocaleModal = ({ isOpen, setIsOpen }) => {
...
 
+  useEffect(() => {
+    /**
+     * @param {number} keyCode - The code of the keyboard key pressed.
+     */
+    function closeFilterOnEscapeKeyPress({ keyCode }) {
+      if (keyCode === 27 && isFiltering) {
+        setIsFiltering(false);
+      }
+    }
```
Comment (jeffchew) [inline packages/react/src/components/LocaleModal/LocaleModal.js:82]: Minor thing, let's make this a private method outside of the `useEffect`, and include the underscore in front of the method name, maybe something like `_closeModalOnEscape`. Be sure to include the `@private` flag in the JSDocs.

File: packages/react/src/components/LocaleModal/LocaleModal.js [hunk 2]
```diff
@@ -162,6 +168,17 @@ const LocaleModal = ({ isOpen, setIsOpen }) => {
   function close() {
     setIsOpen(false);
   }
+
+  /**
+   * @private
```
Comment (jeffchew) [inline packages/react/src/components/LocaleModal/LocaleModal.js:173]: Can you add a description for this private method too?

File: packages/react/src/components/LocaleModal/LocaleModal.js [hunk 3]
```diff
@@ -162,6 +168,17 @@ const LocaleModal = ({ isOpen, setIsOpen }) => {
...
+   *
+   * @param {number} keyCode - The code of the keyboard key pressed.
+   */
+  function _closeFilterOnEscapeKeyPress({ keyCode }) {
```
Comment (jeffchew) [inline packages/react/src/components/LocaleModal/LocaleModal.js:177]: Does this need to be passed in as an object?

Comment (guilhermelMoraes) [reply packages/react/src/components/LocaleModal/LocaleModal.js:177]: The `keyCode` property came from the keydown event, which is an object.

File: packages/react/src/components/LocaleModal/LocaleModal.js (hunk 2 above)
Comment (guilhermelMoraes) [reply packages/react/src/components/LocaleModal/LocaleModal.js:173]: For sure!

File: packages/react/src/components/LocaleModal/LocaleModal.js (hunk 3 above)
Comment (jeffchew) [reply packages/react/src/components/LocaleModal/LocaleModal.js:177]: ok. For JSDocs, we'd want to probably document that then:

```
/**
 * Detects for the esc key press then closes the modal
 * 
 * @param {object} event Event object from the event listener
 * @param {number} event.keyCode code of the keyboard key pressed
 * @private
 */
```

Comment (guilhermelMoraes) [reply packages/react/src/components/LocaleModal/LocaleModal.js:179]: Thanks, Jeff!

**Comment by kennylam** [comment]
@chetankan @jeffchew @guilhermelMoraes Sorry I'm a little late to this but I wasn't aware of #1769. I'm not sure this is the correct interaction. This code is working in the `LocaleModal` story, but will fail in the `Footer` story since `esc` closes the entire modal, which is pretty much the standard behavior for all modals.

We should either trap the modal `esc` while filtering country/lang, or better yet, leave this as it is. This isn't an accessibility issue. What do you guys think?

**Comment by Praveen-sr** [comment]
@jeffchew @guilhermelMoraes Ram re tested this on staging and Learn page, It is working fine. We are able to close the dialog by pressing escape key.

**Comment by guilhermelMoraes** [comment]
> @jeffchew @guilhermelMoraes Ram re tested this on staging and Learn page, It is working fine. We are able to close the dialog by pressing escape key.

@Praveen-sr what exactly is closing when do you press the Esc key? The entire modal or only the filter part?

**Comment by jeffchew** [comment]
@guilhermelMoraes sounds like we should just close this PR.

**Pull request closed without merging**